### PR TITLE
Even better footnotes!

### PR DESCRIPTION
**Pull request by MarcelGerber**
Well, yesterday saw the revamp of footnotes as a side hustle in order to drop jQuery, but today I was at it again and made them even better.

The main news is that they're the new default for mobile (touch) users too, and I think that makes for a way nicer reading experience with less jumping around.
There's a big "Jump to reference" link at the top now so the user is still able to jump to the references section if she wants to.

I also added a max-height to the tooltips and made them scrollable, mostly to make sure they are completely visible.

And then I rewrote the whole thing to use React SSR & hydration to attach the event handlers, so no more calling Tippy directly.

This is what it looks like now on mobile (this is the longest one so it is scrollable):
image

Let me know what you think

### CODE REVIEW

**Comment by danielgavrilov** [comment]
Very awesome to have this @MarcelGerber! 

I went ahead and removed the scroll-on-click completely. I can't think of any case where it's desirable to scroll down to the end of (usually) a very long page to read a bunch of random notes referring to the text.

---

Not ideal that shadows are hiding bits of the scrollbars:

Screenshot 2020-05-27 at 11 23 53

But I feel it's worthwhile for making it more obvious that you can scroll on mobile.

An alternative approach could be adapting `<VerticalScrollContainer>` to work under these circumstances, which would hide the shadows when the user scrolls to the edges. But it's a very small improvement, and the shadows will still show over the scrollbars.

File: site/client/owid.scss [hunk 1]
```diff
@@ -972,30 +1002,23 @@ article.page .authors-byline .citation-note:hover {
...
-    transform: translate(-50%, -50%);
-    content: "increase click area";
-    width: 100%;
-    height: 100%;
-    min-width: 1.875rem;
-    min-height: 1.875rem;
-    z-index: $zindex-footnote;
-    border-radius: 100%;
+    .tippy-content {
+        max-height: 40vh;
```
Comment (danielgavrilov) [inline site/client/owid.scss:1007]: @MarcelGerber Why go for `40vh` here? Isn't it a bit low, even for mobile?

Comment (MarcelGerber) [reply site/client/owid.scss:1007]: The reasoning here is that Tippy can either place the tooltip above or below the ref.
If the ref is exactly in the middle of the screen, then Tippy has only around `50vh` above or below the ref, respectively, to place the tooltip in. For that reason we can't go much higher than `40vw` - 45 would probably be fine, but not much more.

Comment (MarcelGerber) [reply site/client/owid.scss:1007]: That's that scenario:
Screenshot_20200527-125002

**Comment by MarcelGerber** [comment]
Yes, I saw your changes and they work great, too.

For the jump to ref functionality, I looked at how Wikipedia did it, but I totally agree that we shouldn't need it:
image


One thing I noticed, but I think we can live with that, is that tooltips look wacky when they have a scrollbar and the arrow is also on the right (makes the arrow look out of place). That was better when `.tippy-box` had a padding on it, but as I said, definitely not a dealbreaker either.
image

File: site/client/owid.scss (hunk 1 above)
Comment (danielgavrilov) [reply site/client/owid.scss:1007]: Ah, right!

**Comment by MarcelGerber** [comment]
Btw, this looks all good to me, so if you're done and happy with it let's merge it.

**Comment by danielgavrilov** [comment]
It should now look exactly one pixel better! 

I will merge & deploy this tomorrow morning. It's been in our PRs for 3 weeks and I just find it hard to let go at this point.

**Comment by danielgavrilov** [comment]
:rocket:

**Pull request merged**